sequencer/src/api/data_source.rs

```diff
@@ -0,0 +1,33 @@
+use super::endpoints::TimeWindowQueryData;
+use crate::{network, Node, SeqTypes};
+use async_trait::async_trait;
+use hotshot::types::SystemContextHandle;
+use hotshot_query_service::{
+    availability::{AvailabilityDataSource, BlockId},
+    data_source::{UpdateDataSource, VersionedDataSource},
+    status::StatusDataSource,
+    QueryResult,
+};
+
+#[async_trait]
+pub(super) trait SequencerDataSource<N: network::Type>:
+    AvailabilityDataSource<SeqTypes, Node<N>>
+    + StatusDataSource
+    + UpdateDataSource<SeqTypes, Node<N>>
+    + VersionedDataSource
+    + Sized
+{
+    type Options;
+
+    async fn create(opt: Self::Options) -> anyhow::Result<Self>;
+    async fn refresh_indices(&mut self, from_block: usize) -> anyhow::Result<()>;
```

Comment:
Add a bit of documentation to the trait? The types go a long way but it's still not quite clear what for example `refresh_indices` should do.